code/SQLite3Database.php

```
@@ -120,16 +119,6 @@ public static function is_valid_database_name($name)
      */
     public function connect($parameters)
     {
-        if (!empty($parameters['memory'])) {
-            Deprecation::notice(
```

This should throw an InvalidArgumentException instead